### DISCUSSION TITLE
Debian bookworm image for PineTab2

### DISCUSSION THREAD

**Discussion by dieselnutjob** [Plebian-Linux/plebian-website#3] [Ideas]
Hi
I have a very vanilla Debian bookworm image on my PineTab2 and I'm wondering how I can share it to this project.
It's running plasma-mobile and works very nicely either from microSD card or installed onto the internal eMMC
[https://tuxtab.com/20230329_112828.JPG](url)

**Comment by CounterPillow** [comment]
All our images are generated through an automated pipeline, we don't really do one-off artisanally handcrafted images. If you still want them to be part of Plebian, the best option is to start learning [debos](https://github.com/go-debos/debos) to contribute to the pipeline and refactor it into a state where it can generate both types of images (Quartz64 and PineTab2).

There are a few considerations I have with regards to PineTab2: 
- it'll require a custom kernel again, so we need to have some sort of repo with Debian rules files for it to build it with an acceptable versioning scheme so that it can be pushed to our repository. I don't really want to ship a non-repository'd or non-packaged kernel as that doesn't allow for users to update it as easily.
- I need to refactor the Quartz64 image generation pipeline into a few files that get included so that we can more easily customise steps like installing additional packages for some devices
- I can't really test the output of such a hypothetical altered image pipeline on a real PineTab2 right now as it's not out yet, but I'll get one ASAP.

I'm still thinking about how to do various things regarding this, especially how we'll do a kernel package that survives the entirety of Bookworm's release cycle.

For now the best option is to get various parts of PineTab2 support mainlined, namely the device tree and the panel driver. (I know the Wi-Fi driver will need a whole lot more work). This will give me the confidence that I don't have to carry around a kernel package indefinitely for literal years to come and that people will actually try to contribute upstream instead of letting things languish in some personal repository. You should coordinate the mainlining of these things with the author of the respective patches first of course.

**Reply by diederikdehaas** [reply]
I think we should leave Bookworm as is and add support for f.e. PineTab2 to Trixie images.

@dieselnutjob: Can you share what you have wrt building an image (and kernel I presume) for the PineTab2?

**Reply by dieselnutjob** [reply]
I'm using the kernel, dtb and uboot from the Danct12 arch based image with a rootfs created using debootstrap.

**Reply by CounterPillow** [reply]
Unfortunately that's not really useful to us. Any Plebian image I put my name on isn't going to be hacked together from other people's binaries. You're still welcome to distribute your image yourself though, but as I said, Plebian is an automated pipeline and we're not really looking for one-off images.

**Comment by CounterPillow** [comment]
So I think the way forward is the following:
- Plebian will only do Trixie images for PineTab2. This is so that we can hopefully use the vanilla Debian kernel package as things are mainlined. Making a bookworm image for PineTab2 would involve me keeping up with kernel updates for *several years*, which I don't want to do.
  * This means that Plebian won't have a PineTab2 image for months to come as we'll be waiting for a kernel release to include the device tree and panel driver, and then that to trickle down to Debian, and Debian to have moved onto Trixie.
- Device support in the kernel arrives in Plebian by way of having it mainlined and then trickle down to Debian. This encourages all of the various bits to actually get mainlined, instead of putting the maintenance burden on me.
- u-boot support I will get started on once device tree is submitted to kernel. Same u-boot as Plebian, i.e. mainline based. This means no video output, but it also means I don't have to deal with Rockchip's bad ideas.
  * Depending on how RK356x u-boot mainlining works out, I might encourage people to work on features like video output in mainline later down the road. Priority is to mainline all the basic device support first though.
- Image generation pipeline is something I will get started on once Debian Bookworm moves to stable and Trixie becomes Testing. I'll begin refactoring it into smaller pieces that can be re-used before that though.
- Since the Wi-Fi driver is kind of essential but also a very huge minefield right now, I'll consider making a dkms package for it. Ideally we will have a central repository where we can work on it and make it less bad. I'll need to coordinate this with the various involved people.
- The image I ship is unlikely to come with a desktop environment preinstalled, just so we can keep things desktop agnostic. As always though, I'll strongly encourage people to use Wayland based desktops. I might think about having a "user friendly" Plasma based image as things move along, but I feel like others can do this better and provide something like a first-time setup wizard, which Plebian doesn't have.

**Reply by dieselnutjob** [reply]
concerning u-boot.  I have used the pgwipeout quartz64a u-boot on the PT2 and it works just fine (well it boots Linux anyway).
The rest of what you say seems sensible.
